File: 1746-largest-substring-between-two-equal-characters/largest-substring-between-two-equal-characters.java
public class Solution {
    public int maxLengthBetweenEqualCharacters(String s) {
        int c = -1;
        Map<Character, Integer> map = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
            if (!map.containsKey(s.charAt(i))) {
                map.put(s.charAt(i), i);
            } else {
                int x = map.get(s.charAt(i));
                c = Math.max(c, i - x - 1);
            }
        }

        return c;
    }
}

File: 1746-largest-substring-between-two-equal-characters/largest-substring-between-two-equal-characters.cpp
class Solution {
public:
    int maxLengthBetweenEqualCharacters(string s) {
        int i,j,c=-1;

        unordered_map<char,int>map;

        for(i=0;i<s.size();i++){
            if(map.find(s[i])==map.end())
            map[s[i]]=i;
            else{
                int x = map[s[i]];
                c = max(c,i-x-1);
            }
        }

        return c;
    }
};